lib/ansible/modules/utilities/logic/debug.py

```diff
@@ -74,4 +74,10 @@
   debug:
     var: hostvars[inventory_hostname]
     verbosity: 4
+
+# Example that prints two lines of messages, but only if there's an environment value set
+- debug:
+    msg:
+      - "Provisioning based on YOUR_KEY which is: '{{ lookup('env', 'YOUR_KEY') }}"
+      - "It appears you can't do filters though - this '{{ lookup('env', 'YOUR_KEY')|default('all') }}' renders as ''"
```

Comment:
this is incorrect, the problem here is that 'env' returns empty string when the var is not defined, this is not a limitation of debug.